### ISSUE TITLE
Unblinded tokens are empty and unable to view ads in Linux mint

### ISSUE THREAD

**Issue by GeetaSarvadnya** [brave/brave-browser#13336]
## Description 

Unblinded tokens are empty and unable to view ads in Linux mint

## Steps to Reproduce

   1. Clean profile 1.18.77 or 1.19.69 - staging environment
   2. Enable rewards
   3. Try to view an Ad
   4. Ads are not served to the user

## Actual result:

Unblinded tokens are empty and unable to view ads in Linux mint

## Expected result:
Ads should be served to the user

## Reproduces how often: 

Easy
## Brave version (brave://version info)

Brave	1.18.77 Chromium: 87.0.4280.101 (Official Build) (64-bit)
Revision	9407c80213cda69c2b7abcb4fa8e3f74488f4956-refs/branch-heads/4280@{#1807}
OS	Linux

Brave	1.19.69 Chromium: 87.0.4280.101 (Official Build) dev (64-bit)
Revision	9407c80213cda69c2b7abcb4fa8e3f74488f4956-refs/branch-heads/4280@{#1807}
OS	Linux
## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes in 1.19.x
- Can you reproduce this issue with the nightly channel? Not sure

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? NA

## Miscellaneous Information:

CC: @brave/legacy_qa @tmancey @rebron @jsecretan 

In release version `1.18.75` `Brave User Model Installer` is not up-to-date the same issue is there in 1.18.77 also. But `Brave user model Installer` is updated correctly in 1.19.69.

Logs are shared with @tmancey via DM

**Comment by GeetaSarvadnya** [comment]
Please note issue isn't reproducible in Windows 10 x64. Ads can be viewed and the Brave User Model component is also correctly in brave://components in 1.18.x and 1.19.x.  Hence labeled OS/Linux only

**Comment by kjozwiak** [comment]
I couldn't reproduce this on my `Mint 19.1 x64` machine running the following build using both the `staging` and `production` environments:

```
Brave | 1.18.77 Chromium: 87.0.4280.101 (Official Build) (64-bit)
--- | ---
Revision | 9407c80213cda69c2b7abcb4fa8e3f74488f4956-refs/branch-heads/4280@{#1807}
OS | Linux
```

#### **`Staging`**

Once Brave was launched, I was able to redeem all the tokens without any issues, example:

```
[4213:4213:1230/115245.595487:VERBOSE1:refill_unblinded_tokens.cc(270)] Added 50 unblinded tokens, you now have 50 unblinded tokens
[4213:4213:1230/115245.595964:VERBOSE1:ads_impl.cc(538)] Successfully refilled unblinded token
```

I was able to view an ad without any issue as per the following:

```
[4213:4213:1230/115333.245771:VERBOSE1:ad_notification_serving.cc(95)] Ad notification delivered:
  uuid: 82db1f41-913f-4956-9e4b-4e9d23ec53e3
  creativeInstanceId: ed6c467a-8643-42eb-a7cd-32e878fa3cd4
  creativeSetId: a9f0b49c-3298-4c1b-a8b5-76a71e8d7332
  campaignId: 0e70c893-b0c0-4dab-860d-7e65446f0f0f
  category: untargeted
  title: This is a test
  body: This is a test
  targetUrl: https://www.brave.com
```

Once the above ad was displayed, one of the tokens as redeemed and added into the queue as per the following:

```
[4213:4213:1230/115334.182716:VERBOSE1:redeem_unblinded_token.cc(339)] Added 1 unblinded payment token with an estimated redemption value of 0.025 BAT, you now have 1 unblinded payment tokens
[4213:4213:1230/115334.183950:VERBOSE1:ads_impl.cc(512)] Successfully redeemed unblinded token with confirmation id 20a41369-a0f3-4430-a010-7d1a754c2386, creative instance id ed6c467a-8643-42eb-a7cd-32e878fa3cd4 and view
```

**`Example`** | **`Example`**
----------------|----------------
![Screenshot from 2020-12-30 12-10-22](https://user-images.githubusercontent.com/2602313/103369468-359e8e00-4a98-11eb-813a-7ef32ede9af9.png) | ![Screenshot from 2020-12-30 12-10-43](https://user-images.githubusercontent.com/2602313/103369462-320b0700-4a98-11eb-9955-1a165d52832b.png)

**Comment by GeetaSarvadnya** [comment]
@tmancey Could be a regional issue, for @rebron and @kjozwiak it's working fine.

**Comment by kjozwiak** [comment]
> @tmancey Could be a regional issue, for @rebron and @kjozwiak it's working fine.

@GeetaSarvadnya does the same thing happen when you switch over to USA?

**Comment by GeetaSarvadnya** [comment]
>@GeetaSarvadnya does the same thing happen when you switch over to USA? 
Yes

**Comment by tmancey** [comment]
@GeetaSarvadnya could you please run the following from Terminal:

locale

And let me know the results

**Comment by MathInDOS** [comment]
Bruh, Great Browser!

**Comment by MathInDOS** [comment]
https://github.com/MathInDOS

Some of my stuffs...for windows os 😂

**Comment by MathInDOS** [comment]
UI is Great! But it need some improvements for Windows OS. It need to make up like chrome.

**Comment by MathInDOS** [comment]
But Bruh! Brave browser is a legend!

**Comment by GeetaSarvadnya** [comment]
Set the locale to `India` and ran the case again. Ads are not served in `1.19.69` unblinded tokens array is empty although `Brave User Model Installer` is up-to-date in brave://components. Logs are shared via DM

**Comment by tmancey** [comment]
@GeetaSarvadnya @kjozwiak Working as expected for me on both release and beta.

**Comment by tmancey** [comment]
Blocked as does not affect my installation nor @kjozwiak or @LaurenWags

**Comment by GeetaSarvadnya** [comment]
I have done a screen share with @tmancey to investigate the issue, after investigation, we found Linuxmint 17 is quite an older version and has some library/packages missing. Upgraded Linux mint from version 17 to version 17.3 and ran the Ads test again, the issue got resolved, I am gettings ads and unblinded tokens are refilled without any issues. Hence closing the issue. 

![image (6)](https://user-images.githubusercontent.com/38657976/105320661-7481b880-5bec-11eb-8300-a9b6419dbe46.png)
![image](https://user-images.githubusercontent.com/38657976/105320534-48fece00-5bec-11eb-84c6-5285b779d3ee.png)
![image](https://user-images.githubusercontent.com/38657976/105320572-5451f980-5bec-11eb-818b-500a1b255ebc.png)
